src/lab16.py

```diff
@@ -475,20 +476,14 @@ def self_rect(self):
             self.x.get(), self.y.get(), self.x.get() + self.width.get(),
             self.y.get() + self.height.get())
 
-    def is_atomic(self):
-        return not isinstance(self.node, Text) and \
-            (self.node.tag == "input" or self.node.tag == "button")
-
     def paint(self):
         cmds = []
-
-        if not self.is_atomic():
-            bgcolor = self.node.style["background-color"].get()
-            if bgcolor != "transparent":
-                radius = dpx(
-                    float(self.node.style["border-radius"].get()[:-2]),
-                    self.zoom.get())
-                cmds.append(DrawRRect(self.self_rect(), radius, bgcolor))
+        bgcolor = self.node.style["background-color"].get()
```

Comment:
Why did you change this implementation?

Reply:
Ah yeah. This is a pretty dumb bug, and the fix used to be this is_atomic function, but then later we changed how it worked and it is now is supposed to use should_paint. Looks like we never updated this chapter